### PR TITLE
Update WordPress to 5.3

### LINKED ISSUE #177: Upgrade to WordPress 5.3
Need to update Core in the Tide repo which means figuring out what changed in the REST API in 5.0+ that is causing breakage. We are stuck using an old version of Core until we have time to troubleshoot that. It’s going to take some domain knowledge of Tide and REST API to fix it. It would be great if someone figured it out but it’s probably not obvious.

### PR DESCRIPTION
**Pull request by pierlon**
This PR updates the test suite to that of 5.2.3, and also fixes failing tests by getting the instance of the REST server from `rest_get_server` (breaking change occurred at [9e31509](https://github.com/WordPress/wordpress-develop/commit/9e31509293e93383a97ed170039dafa5042ea286) in WP Core.

One of many PRs to close wptide/wptide#177.

### CODE REVIEW

**Comment by derekherman** [comment]
@pierlon This will always fail until you create a branch from the `wptide` repo and not a fork.

**Comment by derekherman** [comment]
Part of the error though is that they are not using proper versioning and it's actually `5.3` for the SVN repo

File: .travis.yml [hunk 1]
```diff
@@ -15,15 +15,15 @@ cache:
 language: php
 
 php:
-  - 7.2.6
+  - 7.3
```
Comment (derekherman) [inline .travis.yml]: This should not be updated. The Docker image this runs on is 7.2.6 which would be a different PR. Please revert this.

File: tests/php/api/controller/class-test-audit-posts-controller.php [hunk 2]
```diff
@@ -14,15 +14,6 @@
...
 
-	/**
-	 * REST Server.
-	 *
-	 * Note that this variable is already defined on the parent class but it lacks the phpdoc variable type.
-	 *
-	 * @var WP_REST_Server
-	 */
-	protected $server;
-
```
Comment (derekherman) [inline tests/php/api/controller/class-test-audit-posts-controller.php:25]: Why remove this and not set the server in the setUp method?

File: tests/php/api/controller/class-test-audit-terms-controller.php [hunk 3]
```diff
@@ -14,15 +14,6 @@
...
 
-	/**
-	 * REST Server.
-	 *
-	 * Note that this variable is already defined on the parent class but it lacks the phpdoc variable type.
-	 *
-	 * @var WP_REST_Server
-	 */
-	protected $server;
-
```
Comment (derekherman) [inline tests/php/api/controller/class-test-audit-terms-controller.php:25]: Never used so makes sense.

File: tests/php/api/endpoint/class-test-report.php [hunk 4]
```diff
@@ -89,7 +89,7 @@ public function setUp() {
 	 * @covers ::register_routes()
 	 */
 	public function test_register_routes() {
-		$routes = $this->server->get_routes();
+		$routes = rest_get_server()->get_routes();
```
Comment (derekherman) [inline tests/php/api/endpoint/class-test-report.php:92]: This should be unchanged and then on line 35 in the setUp method it should be:
```
$this->server = rest_get_server();
```
Which also means removing the global and action which is handled in that function.

File: tests/php/authentication/class-test-jwt-auth.php [hunk 5]
```diff
@@ -91,7 +91,7 @@ function tearDown() {
 	 * @covers ::register_jwt_routes()
 	 */
 	public function test_register_jwt_routes() {
-		$routes = $this->server->get_routes();
+		$routes = rest_get_server()->get_routes();
```
Comment (derekherman) [inline tests/php/authentication/class-test-jwt-auth.php:94]: Same thing here, the setUp method should be modified not this code.

File: tests/php/authentication/class-test-keypair-auth.php [hunk 6]
```diff
@@ -172,7 +172,7 @@ public function test_authenticate_key_pair() {
 	 * @covers ::register_keypair_routes()
 	 */
 	public function test_register_keypair_routes() {
-		$routes       = $this->server->get_routes();
+		$routes       = rest_get_server()->get_routes();
```
Comment (derekherman) [inline tests/php/authentication/class-test-keypair-auth.php:175]: Same thing here, the setUp method should be modified not this code.

File: tests/php/api/controller/class-test-audit-posts-controller.php (hunk 2 above)
Comment (derekherman) [reply tests/php/api/controller/class-test-audit-posts-controller.php:25]: Basically all of the changes to the code below are not needed if the setUp method is changed

File: tests/php/api/endpoint/class-test-report.php (hunk 4 above)
Comment (pierlon) [reply tests/php/api/endpoint/class-test-report.php:92]: The reason I'm using the helper function here (and the other replacements) is because the REST server will be cached and readily available through it. Also, its supposed to offer better [memory cycling](https://github.com/WordPress/wordpress-develop/commit/9e31509293e93383a97ed170039dafa5042ea286).

**Comment by derekherman** [comment]
Closing in favor of #30
